core/src/main/scala/hedgehog/core/PropertyT.scala

```diff
@@ -120,8 +120,12 @@ trait PropertyTReporting[M[_]] {
             case Some((_, None)) =>
               F.map(takeSmallest(ShrinkCount(0), p.config.shrinkLimit, x.map(_._2)))(y => Report(successes, discards, y))
 
-            case Some((m, Some(_))) =>
-              loop(successes.inc, discards, size.inc, x.value._1)
+            case Some((_, Some(_))) =>
+              // Stop looping if the seed was never used - the property never generated anything
+              if (seed == x.value._1)
```

Comment:
Hmm... I'm not 100% sure about this:

>it will still stop if the seed doesn't change.

I think, even if the seed doesn't change, perhaps it should keep going (as-is, right now)? In fact, we might actually want to keep going, even if nothing gets generated, because of erratic/effectful tests.

Perhaps, being explicit by specifying `withTests n` isn't actually a bad idea, for similar reasons that we're explicit with Range (and we should also probably do the same for Size). :thinking:

Reply:
Just thinking out loud is all.